Add support for auto line indentation of multi line config to ConfigRenderer (#445)
eclipse-ankaios/ankaios: doc/docs/usage/manifest/config-rendering.md

# Shareable configuration

> Ankaios supports the shareable configuration approach, which allows configurations to be defined once and assigned to any number of workloads. It uses the [handlebars](https://github.com/sunng87/handlebars-rust) templating syntax to extend certain fields of workloads. For a detailed look at supported workload fields and a basic example of sharing configuration with workloads, see [here](../../reference/startup-configuration.md).

Author: Ok, yes my first approach avoided duplication and linked to the startup-configuration instead. But I think it is good to mention it explicitly here again. It increases usability and you find again all in one place.

            image: ghcr.io/eclipse-ankaios/tests/nginx:alpine-slim
            ports:
            - containerPort: 80
              hostPort: 8080
            volumeMounts:
            - name: nginx-config-volume
              mountPath: /etc/nginx/nginx.conf
              subPath: nginx.conf
          volumes:
          - name: nginx-config-volume
            configMap:
              name: nginx-config
configs:
  nginx_config: |
    worker_processes  1;

    events {
        worker_connections  1024;
    }

    http {
        server {
            listen 80;
            server_name custom_nginx;

            location /custom {
                default_type text/plain;
                return 200 "The mounted custom nginx.conf is being used!\n";
            }
        }
    }
```

With the default template syntax (`{{nginx_conf}}`) instead, the expanded state will result in an invalid YAML:

```yaml
...
    runtimeConfig: |
      manifest: |
        apiVersion: v1
        kind: ConfigMap
        metadata:
          name: nginx-config
        data:
          nginx.conf: |
            worker_processes  1;

events {
    worker_connections  1024;
}

http {
    server {
...
```

> By using the `indent` control structure, incorrect indentation will be prevented and the line indentation of the current context will be considered. This results in an error-free YAML file.

Reviewer: Let's stay positive: 
```suggestion
By using the `indent` control structure, the line indentation of the current context will be considered which results in an error-free YAML file.
```
